packages/identity/src/idp-identity.ts

```diff
@@ -0,0 +1,339 @@
+
+export const defaultPasswordPolicy: PasswordPolicy = {
+  minLength: 8,
+  requireLowercase: true,
+  requireUppercase: true,
+  requireDigits: true,
+  requireSymbols: true,
+  tempPasswordValidity: Duration.days(3),
+};
+
+export interface IdentityProviderProps {
+  readonly [IdentityProviderName.AMAZON]?: UserPoolIdentityProviderAmazonProps;
+  readonly [IdentityProviderName.APPLE]?: UserPoolIdentityProviderAppleProps;
+  readonly [IdentityProviderName.FACEBOOK]?: UserPoolIdentityProviderFacebookProps;
+  readonly [IdentityProviderName.GOOGLE]?: UserPoolIdentityProviderGoogleProps;
+  readonly [IdentityProviderName.OIDC]?: UserPoolIdentityProviderOidcProps;
+  readonly [IdentityProviderName.SAML]?: UserPoolIdentityProviderSamlProps;
+}
+
+/**
+ * Properties which configures the IDP Identity construct.
+ */
+export interface IdpIdentityProps {
+  /**
+   * User provided Cognito UserPool.
+   *
+   * @default - a userpool will be created.
+   */
+  readonly userPool?: IUserPool;
+
+  /**
+   * Props for the UserPool construct
+   */
+  readonly userPoolProps?: UserPoolProps;
+
+  /**
+   * Options while specifying a cognito prefix domain.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-assign-domain-prefix.html
+   */
+  readonly cognitoDomain?: CognitoDomainOptions;
+
+  /**
+   * Options while specifying custom domain
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-add-custom-domain.html
+   */
+  readonly customDomain?: CustomDomainOptions;
+
+  /**
+   * Configuration for the Identity Pool.
+   */
+  readonly identityPoolOptions?: IdentityPoolProps;
+
+  /**
+   * Configuration for the Federated Identity Providers.
+   */
+  readonly identityProviderProps?: IdentityProviderProps;
+
+  /**
+   * Options to create a UserPoolClient
+   */
+  readonly userPoolClientProps?: UserPoolClientOptions;
+
+  readonly identityProviders?: IdentityProviderProps;
+}
+
+/**
+ * Properties which configure a user pool client.
+ */
+export interface AddClientProps extends UserPoolClientOptions {
+  /**
+   * List of allowed redirect URLs for the identity providers.
+   * @default - ['https://example.com'] if either authorizationCodeGrant or implicitCodeGrant flows are enabled, no callback URLs otherwise.
+   */
+  readonly callbackUrls: Array<string>;
+
+  /**
+   * List of allowed logout URLs for the identity providers.
+   * @default - no logout URLs
+   */
+  readonly logoutUrls: Array<string>;
+
+  /**
+   * Options while specifying a cognito prefix domain.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-assign-domain-prefix.html
+   */
+  readonly cognitoDomain?: CognitoDomainOptions;
+
+  /**
+   * Options while specifying custom domain
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-add-custom-domain.html
+   */
+  readonly customDomain?: CustomDomainOptions;
```

Comment:
May be worth noting that you can only specify either `cognitoDomain` or `customDomain`. If it were just typescript you could make it a union type but I imagine jsii would struggle - so best to just document I think.

Reply:
updated with 
```
if (props?.cognitoDomain && props.customDomain) {
      Annotations.of(this).addError('Only one of customDomain or cognitoDomain is allowed');
      throw new Error('Only one of customDomain or cognitoDomain is allowed');
    }
```